src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -243,15 +243,7 @@ internal static TypeSymbolWithAnnotations GetTypeAndNullability(this BoundExpres
                 case BoundKind.Call:
                     return ((BoundCall)expr).Method.ReturnType.IsNullable;
                 case BoundKind.Conversion:
-                    {
-                        var conversion = (BoundConversion)expr;
-                        if (conversion.ExplicitCastInCode)
-                        {
-                            return conversion.IsExplicitlyNullable;
-                        }
-                        Debug.Assert(!conversion.IsExplicitlyNullable);
-                        return null;
-                    }
+                    return ((BoundConversion)expr).ConversionGroup?.ExplicitType?.IsNullable;
```

Comment:
I didn't quite follow this. How do we know that the expression is non-nullable when the conversion group is `null`?

Reply:
Sorry, I still am not following. Why is a the conversion considered oblivious (ie nullability `null`) when it doesn't have a conversion group?

---
In reply to: [189122305](https://github.com/dotnet/roslyn/pull/26356#discussion_r189122305) [](ancestors = 189122305)

Reply:
This method returns the top-level nullability of `expr` in those cases where the nullability can be determined statically. Updated the `ConversionGroup` case and added a comment to the method.

---
In reply to: [189342406](https://github.com/dotnet/roslyn/pull/26356#discussion_r189342406) [](ancestors = 189342406,189122305)